src/Agent/NewRelic/Agent/Parsing/SqlParser.cs

```diff
@@ -357,31 +358,33 @@ protected override ParsedSqlStatement CreateParsedDatabaseStatement(DatastoreVen
         }
 
         /// <summary>
-        ///This method takes any parameters in the given command and plugs them back into the parameterized command text.
+        /// This method takes any parameters in the given command and plugs them back into the parameterized command text.
         /// </summary>
-        public static void FixParameterizedSql(IDbCommand command)
+        /// <param name="command">SQL command to parse</param>
+        /// <returns>True if the plan should be executed.  False if the plan should be aborted.</returns>
+        public static bool FixParameterizedSql(IDbCommand command)
         {
             if (command.Parameters.Count == 0)
             {
-                return;
+                return true; // no params, using raw sql statement.
             }
 
             var sql = command.CommandText;
             foreach (object parameter in command.Parameters)
             {
                 var dbParam = (IDbDataParameter)parameter;
                 //DebugParam(dbParam, sqlObfuscator);
+
+                // Binary: We cannot determine exactly what is in the binary so the out could be invalid for SQL.
+                // Object: Object could be almost anything so there is no easy way ToString it that just works without trying to detect all the differnt objects
+                // It is safer, less error prone, and more efficient to short-circuit on these types than to try and parse them out.
                 var type = dbParam.DbType;
-                object value = dbParam.Value;
-                if (quotableTypes.Contains(type.ToString()))  // the TypeCode for Strings is Int32 for some reason
+                if (type == DbType.Binary || type == DbType.Object)
                 {
-                    value = QuoteString(value.ToString());
+                    Log.DebugFormat("Not executing explain plan since DbType is either Binary or Object."); // not getting the enum string to save resources
```

Comment:
you could get the type and log it by using two if statements with static log messages.

Reply:
I guess we could. I don't really think it needed a separate statement.

Reply:
Split things up.